mteb/evaluation/evaluators/RerankingEvaluator.py

```diff
@@ -129,16 +137,37 @@ def compute_metrics_batched(self, model):
             if num_pos == 0 or num_neg == 0:
                 continue
 
+            for doc in instance["positive"]:
+                qrels[query_id][doc] = 1
+            for doc in instance["negative"]:
+                qrels[query_id][doc] = 0
+
             is_relevant = [True] * num_pos + [False] * num_neg
 
-            scores = self._compute_metrics_instance(query_emb, docs_emb, is_relevant)
+            scores, doc_scores = self._compute_metrics_instance(
+                query_emb, docs_emb, is_relevant
+            )
+            docs = instance["positive"] + instance["negative"]
+
+            for doc, doc_score in zip(docs, doc_scores):
+                relevance_scores[query_id][doc] = doc_score
+
             all_mrr_scores.append(scores["mrr"])
             all_ap_scores.append(scores["ap"])
 
         mean_ap = np.mean(all_ap_scores)
         mean_mrr = np.mean(all_mrr_scores)
 
-        return {"map": mean_ap, "mrr": mean_mrr}
+        evaluator = pytrec_eval.RelevanceEvaluator(qrels, {"ndcg_cut.10"})
```

Comment:
Can we add this section to inside the `_compute_metrics_instance` function just to keep all the metric calculation in one place? You'll have to pass `qrels` into that function, I think.

At some point we should switch over to just using `pytrec_eval.RelevanceEvaluator` for all of them (like the `RetrievalEvaluator` does) but that doesn't have to be this PR.

Also, like the retrieval evaluator can we do multiple nDCG values (not just 10)?  See [this line](https://github.com/embeddings-benchmark/mteb/blob/50c51646dde7aba5bfe6cf66dc9673a61f87a7e6/mteb/evaluation/evaluators/RetrievalEvaluator.py#L464).

Reply:
Yup can do this.